Space -> Project rename & update code-references

develohpanda opened the pull request:
This PR is to rename "Spaces" to "Projects". It also updates code references.

Some things of note:
* the database id prefix has been changed (from `sp` to `proj`)
	* What this means is, while developing locally, if you had previously pulled remote spaces, they would now be duplicated (because the generated id has changed due to the prefix), but this is only going to impact those of us who have been actively developing. If working as an update from 2021.4.1 or a fresh install, users will not see the duplication 
	* I chose `proj` instead of `prj` to at least differentiate from "projects" on the backend (which represent a different entity) <img width="1000" alt="image" src="https://user-images.githubusercontent.com/4312346/130008750-aea09f4d-6458-4c2f-ac09-d8256d28ad3b.png">
* the local storage keys have been updated (both changed keys are new since 2021.4.1)
* the segment events have been updated
* existing mentions of a "project" have been updated to "backendProject"
	* this is because in the current state a "project" in the backend is a collection in the app and not the same entity that is labeled a "Project" in the app - the two represent different things

Aside: yay for using `strings`, most places didn't need any extra change.

<details>
<summary>Screenshots</summary>

<img width="993" alt="image" src="https://user-images.githubusercontent.com/4312346/130004966-4c69c56b-4fe6-44df-8355-13e45798175d.png">

<img width="993" alt="image" src="https://user-images.githubusercontent.com/4312346/130004806-9f48a2ab-5f21-400e-8676-a2027f82e46d.png">

<img width="993" alt="image" src="https://user-images.githubusercontent.com/4312346/130004821-72f3c4f2-7403-48a3-b5c0-a14a951a7f0b.png">

<img width="993" alt="image" src="https://user-images.githubusercontent.com/4312346/130004864-2604532c-ae9c-469d-bf3d-46bb244cb6e5.png">

</details>

develohpanda committed:
update copy and events

develohpanda committed:
change space prefix

develohpanda committed:
change local storage keys

develohpanda committed:
update text in space dropdown

develohpanda committed:
update placeholder

develohpanda committed:
For completeness

develohpanda committed:
change to create locally

develohpanda committed:
update prefix

develohpanda committed:
update test

DMarby commented on packages/insomnia-app/app/common/analytics.ts:
```diff
@@ -121,8 +121,8 @@ export enum SegmentEvent {
   pluginExportLoadWorkspacesInSpace = 'Plugin export loading workspaces for active space',
```
```suggestion
  pluginExportLoadWorkspacesInSpace = 'Plugin export loading workspaces for active project',
```

DMarby commented on packages/insomnia-app/app/ui/redux/modules/__tests__/space.test.js:
```diff
@@ -80,7 +80,7 @@ describe('space', () => {
       } = getAndClearShowAlertMockArgs();
 
       expect(title).toBe('Delete Space');
```
Not a blocker for this PR at all, but there's a bunch of references to `Space` in this file in UI/tracking asserts here still. Since the build still passes, I think something might be wrong with these tests?

develohpanda replied:
Great question, it's possible this wasn't running because it's somehow a JS file. I have updated the file and will make sure the tests run and make sense

develohpanda replied:
Fixed here: 9c73f4fb3f34f7c6593fac6dfa7c97a81bf05783

DMarby commented on packages/insomnia-app/app/models/space.ts:
```diff
@@ -5,7 +5,7 @@ import type { BaseModel } from './index';
 export const name = 'Space';
 export const type = 'Space';
-export const prefix = 'sp';
+export const prefix = 'proj';
```
Does the `name` down in the `init()` function for this model need to be updated as well, or is that never used/shown to the user?

develohpanda replied:
It's never shown, but has also been updated now

DMarby reviewed (dismissed):
:shipit:

dimitropoulos committed:
fully renames space-sort-order

dimitropoulos commented on packages/insomnia-app/app/ui/redux/modules/global.tsx:
```diff
@@ -388,7 +388,7 @@ export const setActiveSpace = (spaceId: string) => {
 
 export const setSpaceSortOrder = (sortOrder: SpaceSortOrder) => {
-  const key = `${LOCALSTORAGE_PREFIX}::space-sort-order`;
+  const key = `${LOCALSTORAGE_PREFIX}::dashboard-sort-order`;
```
In `global.tsx` exists the line:
```ts
export const SET_SPACE_SORT_ORDER = 'global/space-sort-order';
```

Should we, then, change that too?  I fixed it in 389b1a0cbd7719f6e42706476bab0d02d62b7079.

develohpanda replied:
Now that code-references are being updated too, yes 👍🏽

dimitropoulos committed:
finishes rename of `activeSpaceId` to `activeProjectId`

dimitropoulos commented on packages/insomnia-app/app/ui/redux/modules/global.tsx:
```diff
@@ -379,7 +379,7 @@ export const setActiveActivity = (activity: GlobalActivity) => {
 
 export const setActiveSpace = (spaceId: string) => {
-  const key = `${LOCALSTORAGE_PREFIX}::activeSpaceId`;
+  const key = `${LOCALSTORAGE_PREFIX}::activeProjectId`;
```
I feel strongly that we should not let the key in localstorage fall out of sync with what's in the code.  it took me 4 `f12` renames and 50 seconds (I timed it) to rename the remaining `activeProjectId`s.  I pushed that change to 921b434246a299f065a791a222338776bdce0110.  I don't see what we stand to gain by not fixing this now, but I'm open minded to hear something I may not be considering.

develohpanda replied:
Now that code-references are being updated too, this is fine by me 👍🏽

>  it took me 4 f12 renames and 50 seconds (I timed it)

FWIW, my concern with renames is not how long it takes to rename, it's the increased scope of change and testing necessary as a result (esp in this case). Ideally these keys are all constants defined in one place and reused everwhere.

dimitropoulos committed:
renames BASE_PROJECT_ID

dimitropoulos committed:
renames files with `space` in the name

dimitropoulos committed:
renames projectActions

dimitropoulos committed:
renames model and selector variables

dimitropoulos committed:
renames getSpaceId

perhaps important that we do this for ImportRawConfig

dimitropoulos committed:
renames segment event

dimitropoulos committed:
renames createWorkspaceInSpace

dimitropoulos committed:
renames SPACE_SORT_ORDER constant and SpaceSortOrder type

dimitropoulos committed:
updates segment enum and test copy

If this test is not broken, then the code is because `Delete Space` should not appear for users.

dimitropoulos committed:
renames all `spaceId`

dimitropoulos committed:
reames model, type, and string

dimitropoulos committed:
updates askToImportIntoProject

dimitropoulos committed:
updates arguments names space

dimitropoulos committed:
updates user-facing copy message

dimitropoulos committed:
updates all assignments

e.g. all of these are found with a regex like `[^k]space =`

dimitropoulos committed:
renames activeSpace usages

dimitropoulos committed:
adds missing string usage

dimitropoulos committed:
renames forceToSpace

dimitropoulos committed:
renames SpaceDropdown, SpaceDropdownItem, ProjectSortDropdown

dimitropoulos committed:
renames breadcrumb id

dimitropoulos committed:
renames track event category

dimitropoulos committed:
updates test values

dimitropoulos committed:
updates test strings and comments

using a regex like /describe\(.*[^k]space/ and /it\(.*[^k]space/ and ///.*[^k]space/

dimitropoulos committed:
all remaining space useages

as is typical, there were some that I missed, I searched the codebase for `space` and sifted through all the options.

dimitropoulos committed:
renames project to backend project

dimitropoulos committed:
adds file for smoke tests

dimitropoulos committed:
adds space file contents

dimitropoulos committed:
removes space file

develohpanda committed:
adds space file contents

develohpanda committed:
deletes project file

develohpanda committed:
pullBackendProject and initializeBackendProject renames

develohpanda committed:
rename test file

develohpanda committed:
convert test file to ts

develohpanda commented on packages/insomnia-app/app/__jest__/redux-state-for-test.ts:
```diff
@@ -9,8 +9,8 @@ export const reduxStateForTest = async (global: Partial<GlobalState> = {}): Prom
-    activeSpaceId: BASE_SPACE_ID,
-    spaceSortOrder: 'modified-desc',
+    activeProjectId: BASE_PROJECT_ID,
+    projectSortOrder: 'modified-desc',
```
@dimitropoulos I had switched this from `spaceSortOrder` to `dashboardSortOrder` to intentionally decouple the model name, because the sorting is applied to the dashboard, not to the list of projects. I still think this should be `dashboardSortOrder` and will update in a commit, which we can remove should we choose to.

develohpanda replied:
9d377d52e98748b21e024bdf2ccdeb9fc73afc81

develohpanda commented on packages/insomnia-app/app/models/index.ts:
```diff
@@ -42,8 +42,8 @@ export interface BaseModel {
-  //   - Stats, Settings, and Space, which never have a parentId
-  //   - Workspace optionally has a parentId (which will be the id of a Space)
+  //   - Stats, Settings, and Project, which never have a parentId
+  //   - Workspace optionally has a parentId (which will be the id of a project)
```
```suggestion
  //   - Workspace optionally has a parentId (which will be the id of a Project)
```

develohpanda commented on packages/insomnia-app/app/sync/vcs/__tests__/initialize-backend-project.test.ts:
```diff
@@ -5,30 +5,30 @@ import * as models from '../../../models';
 
 const snapshotStateBuilder = createBuilder(snapshotStateEntrySchema);
 
 describe('initialize-project', () => {
```
```suggestion
describe('initialize-backend-project', () => {
```

develohpanda committed:
Update packages/insomnia-app/app/models/index.ts

develohpanda committed:
Update packages/insomnia-app/app/sync/vcs/__tests__/initialize-backend-project.test.ts

develohpanda commented on packages/insomnia-app/app/sync/vcs/vcs.ts:
```diff
@@ -491,11 +491,11 @@ export class VCS {
   }
 
   async shareWithTeam(teamId: string) {
```
Not removing in this PR, but some of these share methods are no longer used and can be safely removed

develohpanda committed:
fix some missing renames

develohpanda committed:
update change code-refs and comments in vcs class

develohpanda committed:
rename remote workspaces hook

develohpanda commented on packages/insomnia-app/app/ui/components/dropdowns/sync-dropdown.tsx:
```diff
@@ -518,7 +518,7 @@ class UnconnectedSyncDropdown extends PureComponent<Props, State> {
             {remoteProjects.length === 0 && (
               <DropdownItem onClick={this._handleEnableSync}>
-                <i className="fa fa-plus-circle" /> Create Local Project
+                <i className="fa fa-plus-circle" /> Create Locally
```
If Insomnia doesn't enable sync automatically (eg. collection imported into a remote space), you have to follow a manual setup sync process (same as currently in the stable release). The first step of this manual setup process contains the terminology "Create Local Project", to initialize a local backend project. What should this be renamed to? For the time being I have renamed to "Create Locally".

** @wdawson should this be "Create Locally" or "Create Local Backend Project" or something else?**

#### This PR
<img width="993" alt="image" src="https://user-images.githubusercontent.com/4312346/130008128-a0b05823-50be-47d0-93ff-7483aa00ffe3.png">

#### 2021.4.1
<img width="992" alt="image" src="https://user-images.githubusercontent.com/4312346/130008202-5e6c1cbe-d67d-40e3-85c1-f9968cb29877.png">

wdawson replied:
I would vote for something like: "Initialize [Collection|Design Document]" if possible. It's confusing to me that the only option would be to do something locally when the whole point of that button is to start a sync. So to me, "initialize" sounds like the first of a few steps and it's reminiscent of a new git repo.

develohpanda replied:
I think the key note here is that it's only initializing locally and that nothing exists on the server until you create and push a snapshot. We can streamline this process a bit more in future to avoid this interim state.

Would a straightforward `Initialize` be sufficient? (This is only available with collections, not documents)

develohpanda commented on packages/insomnia-app/app/ui/components/sidebar/sidebar-request-group-row.tsx:
```diff
@@ -23,7 +23,7 @@ import { SidebarRequestRow } from './sidebar-request-row';
 
 const mapStateToProps = (state: RootState) => ({
-  activeSpace: selectActiveRequest(state),
+  activeProject: selectActiveRequest(state), // TODO this seems wrong (!?!)
```
Woah what the heck, good catch. Seems to not be used at all so at least we're safe on that front :thinking:

(Thanks for marking to change and not changing in this PR :heart:)

develohpanda committed:
revert extra change

develohpanda committed:
more renames

develohpanda committed:
fix tests

develohpanda committed:
Rename to DashboardSortOrder

dimitropoulos committed:
updates from rebase

dimitropoulos committed:
updates test copy

dimitropoulos committed:
fix variable name

dimitropoulos committed:
updates test copy

dimitropoulos committed:
updates action constant (!!!)

how'd we miss this!! lol
